org.eclipse.xtext.xbase/src/org/eclipse/xtext/xtype/util/XFunctionTypeRefs.java

```diff
@@ -0,0 +1,96 @@
+/**
+ * Copyright (c) 2013, 2020 itemis AG (http://www.itemis.eu) and others.
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ * 
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.eclipse.xtext.xtype.util;
+
+import org.eclipse.emf.common.util.URI;
+import org.eclipse.emf.ecore.EObject;
+import org.eclipse.emf.ecore.InternalEObject;
+import org.eclipse.emf.ecore.util.EcoreUtil;
+import org.eclipse.xtext.common.types.JvmParameterizedTypeReference;
+import org.eclipse.xtext.common.types.JvmPrimitiveType;
+import org.eclipse.xtext.common.types.JvmType;
+import org.eclipse.xtext.common.types.JvmTypeReference;
+import org.eclipse.xtext.common.types.JvmVoid;
+import org.eclipse.xtext.common.types.TypesFactory;
+import org.eclipse.xtext.common.types.access.impl.ClassURIHelper;
+import org.eclipse.xtext.xbase.lib.Functions;
+import org.eclipse.xtext.xbase.lib.Procedures;
+
+/**
+ * @author Anton Kosyakov - Initial contribution and API
+ */
+public class XFunctionTypeRefs {
+	public static URI computeTypeUri(boolean procedure, int functionParamCount) {
+		return URI.createURI(XFunctionTypeRefs.buildUri(procedure, functionParamCount));
+	}
+
+	public static String buildUri(boolean procedure, int functionParamCount) {
+		int paramCount = Math.min(6, functionParamCount);
+		if (procedure) {
+			return "java:/Objects/" + Procedures.class.getCanonicalName() + "#" + Procedures.class.getCanonicalName()
```

Comment:
Reminds me of org.eclipse.xtext.common.types.access.impl.ClassURIHelper.getFullURI(Type)